Realm/RLMObjectStore.mm

```diff
@@ -289,19 +289,25 @@ static inline void RLMVerifyInWriteTransaction(RLMRealm *realm) {
 
 static inline void RLMInitializeSwiftListAccessor(RLMObjectBase *object) {
     // switch List<> properties to linkviews from standalone arrays
-    if ([object isKindOfClass:NSClassFromString(@"RealmSwift.Object")]) {
-        for (RLMProperty *prop in object.objectSchema.properties) {
-            if (prop.swiftListIvar) {
+    static Class s_swiftObjectClass = NSClassFromString(@"RealmSwift.Object");
+    if (![object isKindOfClass:s_swiftObjectClass]) {
+        return;
+    }
+
+    for (RLMProperty *prop in object.objectSchema.properties) {
+        if (prop.type == RLMPropertyTypeArray) {
+            static Class s_swiftMigrationObjectClass = NSClassFromString(@"RealmSwift.MigrationObject");
+            if (object.class == s_swiftMigrationObjectClass) {
                 RLMArray *array = [RLMArrayLinkView arrayWithObjectClassName:prop.objectClassName
                                                                         view:object->_row.get_linklist(prop.column)
                                                                        realm:object->_realm];
-                if (object.class == NSClassFromString(@"RealmSwift.MigrationObject")) {
-                    [(id<RLMSwiftMigrationObject>)object initalizeListPropertyWithName:prop.name rlmArray:array];
-                }
-                else {
-                    auto list = static_cast<RLMListBase *>(object_getIvar(object, prop.swiftListIvar));
-                    list._rlmArray = array;
-                }
+                [(id<RLMSwiftMigrationObject>)object initalizeListPropertyWithName:prop.name rlmArray:array];
+            }
+            else {
+                auto list = static_cast<RLMListBase *>(object_getIvar(object, prop.swiftListIvar));
+                list._rlmArray = [RLMArrayLinkView arrayWithObjectClassName:prop.objectClassName
```

Comment:
I think this is fine. Better than doing the conditional twice as we were before. Not enough code to warrant another helper function.